application/utils.py

```
@@ -109,3 +139,42 @@ def local_auth(headers, abort_func):
+    """
+    try:
+        if '@' not in email:
+            return '***@***'
+
+        local, domain = email.split('@', 1)
+        if not local:
+            return '***@' + domain
+
+        masked = local[0] + '*' * (len(local) - 1)
+        return f'{masked}@{domain}'
+    except Exception:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)